lib/ansible/modules/cloud/azure/azure_rm_roledefinition.py

```diff
@@ -0,0 +1,400 @@
+
+        self.scope = self.build_scope()
+
+        # get existing role definition
+        old_response = self.get_roledefinition()
+
+        if old_response:
+            self.results['id'] = old_response['id']
+            self.role = old_response
+
+        if self.state == 'present':
+            # check if the role definition exists
+            if not old_response:
+                self.log("Role definition doesn't exist in this scope")
+
+                self.to_do = Actions.CreateOrUpdate
+
+            else:
+                # existing role definition, do update
+                self.log("Role definition already exists")
+                self.log('Result: {0}'.format(old_response))
+
+                # compare if role definition changed
+                if self.check_update(old_response):
+                    self.to_do = Actions.CreateOrUpdate
+
+        elif self.state == 'absent':
+            if old_response:
+                self.log("Delete role defintion")
+                self.results['changed'] = True
+
+                if self.check_mode:
+                    return self.results
+
+                self.delete_roledefinition(old_response['name'])
+
+                self.log('role definition deleted')
+
+            else:
+                self.log("role definition {0} not exists.".format(self.name))
+
+        if self.to_do == Actions.CreateOrUpdate:
+            self.log('Need to Create/Update role definition')
+            self.results['changed'] = True
+
+            if self.check_mode:
+                return self.results
+
+            response = self.create_update_roledefinition()
+            self.results['id'] = response['id']
+
+        return self.results
+
+    # build scope
+    def build_scope(self):
+        subscription_scope = '/subscription/' + self.subscription_id
+        if self.scope is None:
+            return subscription_scope
+        return self.scope
+
+    # check update
+    def check_update(self, old_definition):
+        if self.description and self.description != old_definition['properties']['description']:
+            return True
+        if self.permissions:
+            if len(self.permissions) != len(old_definition['permissions']):
+                return True
+            existing_permissions = self.permissions_to_set(old_definition['permissions'])
+            new_permissions = self.permissions_to_set(self.permissions)
+            if existing_permissions != new_permissions:
+                return True
+        if self.assignable_scopes and self.assignable_scopes != old_definition['assignable_scopes']:
+            return True
+        return False
+
+    def permissions_to_set(self, permissions):
+        new_permissions = [str(dict(
+            actions=(set([to_native(a) for a in item.get('actions')]) if item.get('actions') else None),
+            non_actions=(set([to_native(a) for a in item.get('non_actions')]) if item.get('non_actions') else None),
```

Comment:
Shouldn't these be `not_actions` ? ... and the tests should probably include some of these to catch the fact that they're not working... ;)